### PR TITLE
bump gradle to 7.5.1

### PR DESCRIPTION
**Pull request by alexander-kulikovskii**
### Summary of changes

Update gradle version to [7.5.1](https://docs.gradle.org/7.5.1/release-notes.html).
Also it's unblock us in using new dependency system.

## Pull request checklist:
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [ ] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-all.zip
```
Comment (pengdev) [inline gradle/wrapper/gradle-wrapper.properties:6]: would be good to validate our gradle plugins is working correctly with this gradle version.

**Review by kiryldz — COMMENTED** [review]
Can you please add PR description? Are we updating gradle to unblock some things?

**Comment by alexander-kulikovskii** [comment]
> Can you please add PR description? Are we updating gradle to unblock some things?

done

**Comment by yunikkk** [comment]
> new dependency system.

Not sure, what dependency system is meant?

**Comment by alexander-kulikovskii** [comment]
> > new dependency system.
> 
> Not sure, what dependency system is meant?

https://proandroiddev.com/gradle-version-catalogs-for-an-awesome-dependency-management-f2ba700ff894

**Comment by jush** [comment]
Have the Gradle custom plugins been tested? (e.g. SDK Registry one)

**Review by jush — APPROVED** [review]

File: scripts/checksum.sh [hunk 2]
```diff
@@ -23,7 +23,8 @@ checksum_file() {
 }
 
 add_files . build.gradle.kts
-add_files ./buildSrc/src/main/kotlin/ Project.kt
+add_files ./buildSrc/src/main/kotlin Project.kt
+add_files ./gradle/wrapper gradle-wrapper.properties
```
Comment (kiryldz) [inline scripts/checksum.sh:27]: just a bit curious why adding it here now

**Pull request merged**